specs/agents/sanitization.md

```diff
@@ -1,16 +1,58 @@
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to
+be interpreted as described in
+[RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
 ## Data sanitization
 
 ### `sanitize_field_names` configuration
 
 Sometimes it is necessary to sanitize, i.e., remove,
 sensitive data sent to Elastic APM.
-This config accepts a list of wildcard patterns of field names which should be sanitized.
-These apply to HTTP headers (including cookies) and `application/x-www-form-urlencoded` data (POST form fields).
-The query string and the captured request body (such as `application/json` data) will not get sanitized.
+
+This config accepts a list of wildcard patterns of field names which control
+how an agent will sanitize data.
 
 |                |   |
 |----------------|---|
 | Type           | `List<`[`WildcardMatcher`](../../tests/agents/json-specs/wildcard_matcher_tests.json)`>` |
 | Default        | `password, passwd, pwd, secret, *key, *token*, *session*, *credit*, *card*, authorization, set-cookie` |
 | Dynamic        | `true` |
 | Central config | `true` |
+
+## Configuration
+
+Agents MUST provide a minimum default configuration of
+
+    [ 'password', 'passwd', 'pwd', 'secret', '*key', '*token*', '*session*',
+      '*credit*','*card*', 'authorization', 'set-cookie']
+
+for the `sanitize_field_names` configuration value.  Agent's MAY include the
+following extra fields in their default configuration to avoid breaking changes
+
+    ['pw','pass','connect.sid']
```

Comment:
The fact that some agents have different defaults should be reflected in [the remote config description](https://github.com/elastic/kibana/blob/7.x/x-pack/plugins/apm/common/agent_configuration/setting_definitions/general_settings.ts#L227).

Reply:
Today I learned there's a remote config description :) These extra fields were/are for the Node.js agent specifically -- they're fields that we previously redacted that would be tricky/cumbersome (but not impossible) to change.  However, since it sounds like there's downstream dependencies I'm learning towards just removing this and going with the defaults. 

@eyalkoren (or anyone) A few follow up questions for you -- is the the remote config description _used_ by the central configuration system?  

That is -- with those defaults set in `general_settings.ts` does that mean the central configuration system will send these values by default?  The follow on question that is -- do agents typically need to set these default values themselves, or can/should we rely on them being set by central configuration?

Reply:
>  is the the remote config description used by the central configuration system?

No, it is only a string for description of the default.

> does that mean the central configuration system will send these values by default? The follow on question that is -- do agents typically need to set these default values themselves, or can/should we rely on them being set by central configuration?

The central config does not send these values, agents need to set them themselves. After all, the central config is just one more configuration system for agents, which need to have proper defaults even if it is disabled, and you would typically use the same default regardless of the configuration source.

Reply:
@astorm So you leaning towards having the apm-agent-nodejs *drop* `['pw','pass','connect.sid']` by default?  I don't disagree. We'll need to defer doing so until a major ver bump, right?

Reply:
Thank you @eyalkoren -- the description of this issue has been updated with a TO DO that reflects the need to update the remote config description.

Reply:
@trentm I suspect we'll talk off of GitHub about this, but my intent was to spec `['pw','pass','connect.sid']` this way in order to give us some flexibility in keeping these extra keys redacted and not force a major version bump on us immediately.